remote/mdb_io.go

```diff
@@ -283,3 +284,88 @@ type loggingCacheLenRequest struct {
 type loggingSendMessageRequest struct {
 	Payload options.LoggingPayload `bson:"send_message"`
 }
+
+type scriptingCreateRequest struct {
+	Params struct {
+		Type    string `bson:"type"`
+		Options []byte `bson:"options"`
```

Comment:
Can't this just store the `options.ScriptingHarness` interface itself rather than send as bytes? `shell.RequestToMessage` already calls `bson.Marshal`.

Reply:
You can't bson encode an interface but I can bson encode type `interface{}`. The problem then is that I would need to do a type conversion to the underlying type on the server side to get it back to the `options.ScriptingHarness`.

Reply:
I think this is fair but it might be more cumbersome for end users to have to serialize their inputs to raw BSON bytes before being able to send a scripting request. All of the requests were written in such a way that you could look at the struct definition and determine what the arguments were without needing to do some data transformation on the input. If Options was an `interface{}` type it would accept anything as long as it could be type casted into one of the scripting option types since they already have to specify the type anyways.